test/Sync.t.sol

```diff
@@ -0,0 +1,244 @@
+
+        uint256 balance = manager.sync(currency0);
+        assertEq(balance, currency0Balance, "balance not equal");
+        assertEq(manager.getReserves(currency0), balance);
+    }
+
+    function test_settle_withStartingBalance() public {
+        assertGt(currency0.balanceOf(address(manager)), uint256(0));
+
+        IPoolManager.SwapParams memory params =
+            IPoolManager.SwapParams({zeroForOne: true, amountSpecified: -100, sqrtPriceLimitX96: SQRT_RATIO_1_2});
+
+        PoolSwapTest.TestSettings memory testSettings =
+            PoolSwapTest.TestSettings({withdrawTokens: true, settleUsingTransfer: true});
+
+        // Sync has not been called.
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        manager.getReserves(currency0);
+
+        swapRouter.swap(key, params, testSettings, new bytes(0));
+        (uint256 balanceCurrency0) = currency0.balanceOf(address(manager));
+        assertEq(manager.getReserves(currency0), balanceCurrency0); // Reserves are up to date since settle was called.
+    }
+
+    function test_settle_withNoStartingBalance() public {
+        assertEq(currency2.balanceOf(address(manager)), uint256(0));
+
+        (Currency cur0, Currency cur1) = currency0 < currency2 ? (currency0, currency2) : (currency2, currency0);
+        PoolKey memory key2 =
+            PoolKey({currency0: cur0, currency1: cur1, fee: 3000, tickSpacing: 60, hooks: IHooks(address(0))});
+
+        manager.initialize(key2, SQRT_RATIO_1_1, new bytes(0));
+
+        // Sync has not been called.
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        manager.getReserves(currency2);
+        modifyLiquidityRouter.modifyLiquidity(key2, IPoolManager.ModifyLiquidityParams(-60, 60, 100), new bytes(0));
+        (uint256 balanceCurrency2) = currency2.balanceOf(address(manager));
+        assertEq(manager.getReserves(currency2), balanceCurrency2);
+    }
+
+    function test_settle_revertsIfSyncNotCalled() public {
+        Actions[] memory actions = new Actions[](1);
+        bytes[] memory params = new bytes[](1);
+
+        actions[0] = Actions.SETTLE;
+        params[0] = abi.encode(currency0);
+
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        router.executeActions(actions, params);
+    }
+
+    /// @notice When there is no balance and reserves are set to type(uint256).max, no delta should be applied.
+    function test_settle_noBalanceInPool_shouldNotApplyDelta() public {
+        assertEq(currency2.balanceOf(address(manager)), uint256(0));
+
+        // Sync has not been called.
+        vm.expectRevert(Reserves.ReservesMustBeSynced.selector);
+        manager.getReserves(currency2);
+
+        manager.sync(currency2);
+        assertEq(manager.getReserves(currency2), 0);
+
+        Actions[] memory actions = new Actions[](2);
+        actions[0] = Actions.SETTLE;
+
+        bytes[] memory params = new bytes[](2);
+        params[0] = abi.encode(currency2);
```

Comment:
super super nit but i would rather this test was like the others and did
```
Actions[] memory actions = new Actions[](2);
bytes[] memory params = new bytes[](2);
```

and then

```
        actions[0] = Actions.SETTLE;
        params[0] = abi.encode(currency2);
```

much easier to read what the first action is that way